src/client/dfs/dfs.c

```diff
@@ -1562,6 +1573,16 @@ dfs_mkdir(dfs_t *dfs, dfs_obj_t *parent, const char *name, mode_t mode,
 
 	strncpy(new_dir.name, name, DFS_MAX_PATH);
 	new_dir.name[DFS_MAX_PATH] = '\0';
+
+	if (dfs->use_dtx) {
+		rc = daos_tx_open(dfs->coh, &th, 0, NULL);
+		if (rc) {
+			D_ERROR("daos_tx_open() failed (%d)\n", rc);
+			D_GOTO(out, rc);
+		}
+	}
+
+restart:
 	rc = create_dir(dfs, th, parent->oh, cid, &new_dir);
```

Comment:
Looks like create_dir doesn't need th.

Reply:
ah yes i can remove that from the create_dir args.
but still need tx for the insert_entry.